### ISSUE TITLE
Would required positional arguments has a "required" option?

### ISSUE THREAD

**Issue by HairyRabbit** [yargs/yargs#1393]
Hi, some questions. :)

I write my cli by typescript:

```ts
yargs.command(
  `$0 <foo>`, 'desc', 
  yargs => {
    yargs.positional('foo', { type: 'string' })
  },
  args => {
    const { _, $0, foo } = args
    // foo type was "string | undefined"
  }
)
```

Add the default value `{ type: 'string', default: '' }` was fine. Is there other way let foo type not include `undefined`? because the positional argument already required.

**Comment by mleguen** [comment]
Did you try adding `demandOption: "true"` to the positional?

I think this is the solution I found to a similar issue a few months ago. But you must provide `"true"` as a string, not a boolean, for the `undefined` to be removed from the typescript type.

**Comment by bcoe** [comment]
@mleguen this sounds like a bug with the type definitions?

**Comment by HairyRabbit** [comment]
@mleguen  
Thinks, the solution works for me. I checked type definition file:

```ts
type InferredOptionType<O extends Options | PositionalOptions> =
        O extends { default: infer D } ? D :
        O extends { type: "count" } ? number :
        O extends { count: true } ? number :
        O extends { required: string | true } ? RequiredOptionType<O> :
        O extends { require: string | true } ? RequiredOptionType<O> :
        O extends { demand: string | true } ? RequiredOptionType<O> :
        O extends { demandOption: string | true } ? RequiredOptionType<O> :
        RequiredOptionType<O> | undefined;
```

Looks like literal type `true` doesn't work. Any other string or change `true` to `boolean` are works.

**Comment by mleguen** [comment]
@bcoe Yes, this is a bug with the type definitions in `@types/yargs` I found a few months ago and forgot to open an issue about.

**Comment by douniwan5788** [comment]
none of the `required、require、demand、demandOption` works for `positional` no matter boolean or string `true`

**Comment by ivan-kleshnin** [comment]
> none of the required、require、demand、demandOption works for positional no matter boolean or string true

Not completely true. 
- `<cmd>` syntaxl affects JS behaviors.
- `demandOption` affects TypeScript type, e.g. `string | undefined` -> `string`.

**Comment by ThePlenkov** [comment]
Seems like this one works:
```
const argv = yargs(hideBin(process.argv))
  .command('<argument>', 'Positional argument')
  .demandCommand(1)
  .parseSync();
```

Taken from [here](https://remarkablemark.org/blog/2021/10/16/yargs-command-with-positional-argument/)
